[CodeCompletion] Don't attach attr to decl if blank line exists

rintaro opened the pull request:
```swift
@#^COMPLETE^#

public func something() {}
```
In this case, we can't say the user is adding attribute to the `func` or starting a new declaration. So if there're one or more blank lines after the completion, suggest context free attribute list.

rdar://problem/50441643

rintaro committed:
[CodeCompletion] Don't attach attr to decl if blank line exists

```
@#^COMPLETE^#

public func something() {}
```
In this case, we can't say the user is adding attribute to the func or
starting a new declaration. So if there're one or more blank lines after the
completion, suggest context free attribute list.

rdar://problem/50441643

rintaro commented:
@swift-ci Please smoke test

benlangmuir commented on lib/IDE/CodeCompletion.cpp:
```diff
@@ -1351,6 +1351,13 @@ class CodeCompletionCallbacksImpl : public CodeCompletionCallbacks {
+    // If the target is already set to 'Module', that means the completion
+    // should be performed as if it's not tied to any specific decl.
+    // see 'completeDeclAttrBeginning()'
+    if (AttTargetDK == DeclKind::Module)
```
I'd prefer we just store the "isindependent" bit, maybe as "AttTargetIsIndependent" so we can just use `None` instead of this dance with `Module`.

benlangmuir reviewed (commented):
I think we should do this even if the decl is on the next line (no blank line in between).  It's common to have decls on consecutive lines.  Better to show too many attributes than too few.

rintaro committed:
[CodeCompletion] Stop using DeclKind::Module as a isIndependent indicator

rintaro committed:
[CodeCompletion] contextual attribute completion only on the same line

It's common to have decls on consecutive lines. Better to show too many
attributes than too few.

rintaro commented:
Modified to suggest context-free attribute list if the next token is `isAtStartOfLine()`.
@benlangmuir How does this look?

rintaro commented:
@swift-ci Please smoke test

benlangmuir commented on lib/IDE/CodeCompletion.cpp:
```diff
@@ -5373,13 +5379,16 @@ void CodeCompletionCallbacksImpl::doneParsing() {
-    Lookup.getAttributeDeclCompletions(IsInSil, AttTargetDK);
+    auto DK = AttTargetDK;
+    if (AttTargetIsIndependent)
+      DK = None;
```
Why do we check this here instead of in `setAttrTargetDeclKind` ?

rintaro replied:
No strong reason. Moved to `setAttrTargetDeclKind `.

rintaro commented:
@swift-ci Please smoke test
